src/balancerd/src/lib.rs

```diff
@@ -622,16 +633,19 @@ impl mz_server_core::Server for HttpsBalancer {
     const NAME: &'static str = "https_balancer";
 
     fn handle_connection(&self, conn: TcpStream) -> mz_server_core::ConnectionHandler {
-        let tls_context = Arc::clone(&self.tls);
+        let tls_context = self.tls.clone();
         let resolve_template = Arc::clone(&self.resolve_template);
         let metrics = self.metrics.clone();
         Box::pin(async move {
             let active_guard = metrics.active_connections();
             let result = Box::pin(async move {
                 let (mut client_stream, servername): (Box<dyn ClientStream>, Option<String>) =
-                    match &*tls_context {
+                    match tls_context {
                         Some(tls_context) => {
-                            let mut ssl_stream = SslStream::new(Ssl::new(tls_context)?, conn)?;
+                            let mut ssl_stream = SslStream::new(
+                                Ssl::new(&tls_context.context.read().expect("poisoned"))?,
```

Comment:
Suggest making `ReloadingSslContext::context` a private field, and adding a public `.get()` accessor method that handles the `.context.read().expect("poisoned")` bit, just to keep things a bit more abstracted.

```suggestion
                                Ssl::new(&tls_context.context.get())?,
```